lib/unifex/code_generators/nif.ex

```diff
@@ -456,4 +467,58 @@ defmodule Unifex.CodeGenerators.NIF do
   defp generate_struct_native_definition(struct_data, ctx) do
     Unifex.CodeGenerators.Common.generate_struct_native_definition(struct_data, NIF, ctx)
   end
+
+  # This function generates bugged version of functions declarations returning nil and should be removed in v2.0.0
+  defp generate_functions_declarations_bugged(config, ctx) do
+    config
+    |> Enum.map(fn {name, result} ->
+      {_result, meta} = generate_serialization(result, ctx)
+      args = meta |> Keyword.get_values(:arg)
+
+      args_declarations =
+        [~g<UnifexEnv* env> | generate_args_declarations(args, :const_unless_ptr_on_ptr, ctx)]
+        |> Enum.join(", ")
+
+      labels =
+        meta
+        |> Keyword.get_values(:label)
+
+      if Enum.member?(labels, "nil") do
+        ~g<UNIFEX_TERM #{[name, :result, ""] |> Enum.join("_")}(#{args_declarations})>
+      else
+        ~g<>
+      end
+    end)
+    |> List.flatten()
+    |> Enum.filter(&(&1 != ""))
+    |> Enum.map_join("\n", &(&1 <> ";"))
+  end
+
+  # This function generates bugged version of functions returning nil and should be removed in v2.0.0
+  defp generate_functions_bugged(config, ctx) do
+    config
+    |> Enum.map(fn {name, result} ->
+      {_result, meta} = generate_serialization(result, ctx)
+
+      labels =
+        meta
+        |> Keyword.get_values(:label)
+
+      if Enum.member?(labels, "nil") do
+        ~g"""
+        #{args = meta |> Keyword.get_values(:arg)
+
+        args_declarations = [~g<UnifexEnv* env> | generate_args_declarations(args, :const_unless_ptr_on_ptr, ctx)] |> Enum.join(", ")
+
+        ~g<UNIFEX_TERM #{[name, :result, ""] |> Enum.join("_")}(#{args_declarations})>} {
+          return #{BaseType.generate_arg_serialize(:atom, :"\"#{nil}\"", NIF, ctx)};
+        }
+        """
+      else
+        ~g<>
+      end
+    end)
+    |> Enum.filter(&(&1 != ""))
```

Comment:
Instead of combining `Enum.map/2` and `Enum.filter/2` you can just use `Enum.flat_map/2` here in similar places. It will simplify your code

Reply:
Unchanged